packages/mobile/src/components/Files/ChatFileOverview.tsx

```diff
@@ -124,45 +134,72 @@ const FilePreview = ({
         </View>
       ) : (
         <View style={[style]}>
-          <Image source={{ uri: asset.uri || asset.originalPath }} style={[size, imageStyle]} />
+          <Image
+            source={{ uri: asset.uri || asset.filepath || undefined }}
+            style={[size, imageStyle]}
+          />
           {children}
         </View>
       )}
     </>
   );
 };
+export type ChatFileOverviewProp = NativeStackScreenProps<ChatStackParamList, 'ChatFileOverview'>;
 
-// TODO: Perhaps convert this to a route
 export const ChatFileOverview = memo(
   ({
-    title,
-    assets,
-    setAssets,
-    doSend,
-  }: {
-    title?: string;
-    assets: Asset[];
-    setAssets: React.Dispatch<React.SetStateAction<Asset[]>>;
-    doSend: (message: { text: string }[]) => void;
-  }) => {
+    navigation,
+    route: {
+      params: { initialAssets, recipients },
+    },
+  }: ChatFileOverviewProp) => {
     const [messageInput, setIsMessageInput] = useState(false);
     const [message, setMessage] = useState('');
     const { isDarkMode } = useDarkMode();
+    const [assets, setAssets] = useState(initialAssets);
     const [currentIndex, setCurrentIndex] = useState(0);
     const currentAsset = assets[currentIndex];
     const { bottom: bottomInsets } = useSafeAreaInsets();
     const { colors } = useTheme();
+    const { mutate: sendMessage } = useChatMessage().send;
+
+    const onSend = useCallback(() => {
+      recipients.forEach((recipient) => {
+        sendMessage({
```

Comment:
Perhaps add some error handling as well? A async `SendMessage` might make more sense, so you can catch any errors, and display an error to the user

Reply:
I thought about this. async `SendMessage` would not make sense as it would wait for every request to be completed before popping up the screen. This is fine when sending to a single recipient but when multiple dealing with multiple recipients and multiple images, this may take a very long time. If any request fails, it would still be visible in the respective chat screen right?

Reply:
Well, if it's multiple recipients, everyone will assume they all got sent correctly, as you show a message "Message sent successfully" but you don't actually check before showing that 😉 ; So by await'ing all together, they can run async, but at least the success state is only shown when they all finished

For a single recipient, it might make sense to not await, but even then I think it can be reasonable to let the user await it, but that's your call.